packages/next-app/src/app/internalmanager/page.tsx

```diff
@@ -188,3 +153,100 @@ export default function Page() {
     </div>
   );
 }
+
+function TableRow({
+  token,
+  userAddress,
+  setTransaction,
+  chain,
+}: {
+  token: {
+    __typename?: "UserInternalBalance" | undefined;
+    token: `0x${string}`;
+    balance: string;
+  };
+  userAddress: `0x${string}`;
+  setTransaction: React.Dispatch<React.SetStateAction<ITransaction>>;
+  chain?: Chain;
+}) {
+  const [transferKind, setTransferKind] = useState<UserBalanceOpKind>();
+  const userBalanceOp = {
+    kind: transferKind as number,
+    asset: token.token,
+    amount: parseFixed(token.balance, tokenDictionary[token.token].decimals),
+    sender: userAddress as `0x${string}`,
+    recipient: userAddress as `0x${string}`,
+  };
+  const { config } = usePrepareVaultManageUserBalance({
+    args: [[userBalanceOp]],
+  });
+  const { data, write } = useVaultManageUserBalance(config);
+
+  function handleWithdraw() {
+    setTransferKind(UserBalanceOpKind.WITHDRAW_INTERNAL);
+    setTransaction((prev) => ({
+      ...prev,
+      status: TransactionStatus.WAITING_APPROVAL,
+    }));
+  }
+
+  useEffect(() => {
+    if (!data) return;
```

Comment:
could add `tiny-invariant` for these checks

Reply:
hmmmm not quite sure if `tiny-invariant` do what we want here. I just want to check if data exists before trying to manipulate data. But if it doesn't exist I just want to skip the rest of the function for now as this is an useEffect.  If I use `tiny-invariant` it'll throw the error on the first rendering as data is still undefined

Reply:
question is what would cause the `data` to not exist then - is the wagmi generated hook a promise? trying to generate it locally but getting an error

Reply:
By default, the `useContractWrite()` from wagmi core has data as undefined unless a transaction is occurring
<img width="542" alt="image" src="https://user-images.githubusercontent.com/63298341/232111544-8fb37bbf-ae49-465f-b825-43c42edff2b0.png">

Reply:
hmm ok, thought we should revert if we don't have `data` but it behaves differently 🤔 still thinking about this useEffect though, wagmi should handle it natively but if we don't find a better solution that's fine too.